fix from Chet src: https://lists.gnu.org/archive/html/bug-bash/2020-04/msg00063.html
for bash 'devel' branch at commit b3968b54c13def5803efc82d4c63d84446fe28bf

#*** bash-20200408/lib/readline/display.c	2020-04-07 14:55:15.000000000 -0400
#--- fix/lib/readline/display.c	2020-04-11 11:46:04.000000000 -0400
#***************
#*** 1758,1763 ****
#--- 1758,1766 ----
#  		  ne = new + nmax;
#  		  nd = newbytes;
#  		  nfd = new + nd;
#+ 		  ofdf = old_face;
#+ 		  nfdf = new_face;
#+ 
#  		  goto dumb_update;
#  		}
#  	      if (oldbytes != 0 && newbytes != 0)
#
#more correct version src: https://lists.gnu.org/archive/html/bug-bash/2020-04/msg00066.html
#ok this fails to apply likely because copy/paste didn't see tabs!!:
#*** bash-20200408/lib/readline/display.c     2020-04-07 14:55:15.000000000 
#-0400
#--- fix/lib/readline/display.c      2020-04-12 12:01:22.000000000 -0400
#***************
#*** 1759,1762 ****
#--- 1759,1765 ----
#                  nd = newbytes;
#                  nfd = new + nd;
#+                 ofdf = old_face + oldbytes;
#+                 nfdf = new_face + newbytes;
#+
#                  goto dumb_update;
#                }
#
#
diff --git a/lib/readline/display.c b/lib/readline/display.c
index 14af1e92..407d1c3c 100644
--- a/lib/readline/display.c
+++ b/lib/readline/display.c
@@ -1758,6 +1758,9 @@ update_line (char *old, char *old_face, char *new, char *new_face, int current_l
 		  ne = new + nmax;
 		  nd = newbytes;
 		  nfd = new + nd;
+		  ofdf = old_face + oldbytes;
+		  nfdf = new_face + newbytes;
+
 		  goto dumb_update;
 		}
 	      if (oldbytes != 0 && newbytes != 0)
